tools/qemu-kvm-patch/kvm-qemu-2.4.0.1-libvmi.patch

```diff
@@ -0,0 +1,372 @@
+@@ -905,6 +905,15 @@
+     hmp_handle_error(mon, &err);
+ }
+ 
++void hmp_pmemaccess(Monitor *mon, const QDict *qdict)
++{
++    const char *path = qdict_get_str(qdict, "path");
++    Error *err = NULL;
++
++    qmp_pmemaccess(path, &err);
++    hmp_handle_error(mon, &err);
++}
++
+ void hmp_ringbuf_write(Monitor *mon, const QDict *qdict)
+ {
+     const char *chardev = qdict_get_str(qdict, "device");
+diff -Naur qemu-2.4.0.1/hmp-commands.hx qemu-2.4.0.1-valerio/hmp-commands.hx
+--- qemu-2.4.0.1/hmp-commands.hx	2015-09-22 16:00:48.000000000 -0600
++++ qemu-2.4.0.1-valerio/hmp-commands.hx	2015-10-12 10:31:02.811815385 -0600
+@@ -805,6 +805,20 @@
+ ETEXI
+ 
+     {
++        .name       = "pmemaccess",
++        .args_type  = "path:s",
++        .params     = "file",
++        .help       = "open A UNIX Socket access to physical memory at 'path'",
++        .mhandler.cmd = hmp_pmemaccess,
++    },
++
++STEXI
++@item pmemaccess @var{file}
++@findex pmemaccess
++Mount guest physical memory image at 'path'
++ETEXI
++
++    {
+         .name       = "boot_set",
+         .args_type  = "bootdevice:s",
+         .params     = "bootdevice",
+diff -Naur qemu-2.4.0.1/hmp.h qemu-2.4.0.1-valerio/hmp.h
+--- qemu-2.4.0.1/hmp.h	2015-09-22 16:00:48.000000000 -0600
++++ qemu-2.4.0.1-valerio/hmp.h	2015-10-12 10:15:52.583843591 -0600
+@@ -46,6 +46,7 @@
+ void hmp_cpu(Monitor *mon, const QDict *qdict);
+ void hmp_memsave(Monitor *mon, const QDict *qdict);
+ void hmp_pmemsave(Monitor *mon, const QDict *qdict);
++void hmp_pmemaccess(Monitor *mon, const QDict *qdict);
+ void hmp_ringbuf_write(Monitor *mon, const QDict *qdict);
+ void hmp_ringbuf_read(Monitor *mon, const QDict *qdict);
+ void hmp_cont(Monitor *mon, const QDict *qdict);
+diff -Naur qemu-2.4.0.1/Makefile.target qemu-2.4.0.1-valerio/Makefile.target
+--- qemu-2.4.0.1/Makefile.target	2015-09-22 16:00:47.000000000 -0600
++++ qemu-2.4.0.1-valerio/Makefile.target	2015-10-12 10:15:52.583843591 -0600
+@@ -128,7 +128,7 @@
+ #########################################################
+ # System emulator target
+ ifdef CONFIG_SOFTMMU
+-obj-y += arch_init.o cpus.o monitor.o gdbstub.o balloon.o ioport.o numa.o
++obj-y += arch_init.o cpus.o monitor.o gdbstub.o balloon.o ioport.o numa.o memory-access.o
+ obj-y += qtest.o bootdevice.o
+ obj-y += hw/
+ obj-$(CONFIG_KVM) += kvm-all.o
+diff -Naur qemu-2.4.0.1/memory-access.c qemu-2.4.0.1-valerio/memory-access.c
+--- qemu-2.4.0.1/memory-access.c	1969-12-31 17:00:00.000000000 -0700
++++ qemu-2.4.0.1-valerio/memory-access.c	2015-10-12 10:21:06.155833874 -0600
+@@ -0,0 +1,206 @@
++/*
++ * Access guest physical memory via a domain socket.
++ *
++ * Copyright (C) 2011 Sandia National Laboratories
++ * Original Author: Bryan D. Payne (bdpayne@acm.org)
++ * 
++ * Refurbished for modern QEMU by Valerio Aimale (valerio@aimale.com), in 2015
++ */
++
++#include "memory-access.h"
++//#include "cpu-all.h"
++#include "qemu-common.h"
++#include "exec/cpu-common.h"
++#include "config.h"
```

Comment:
Would rather include `qemu/osdep.h` for compatibility with 2.6+, as config.h has been removed recently.

I believe I might be getting the same issue as #333 when I interrupt any of the examples and try to run another though. Seems to create another socket but libvmi not being able to see it for whatever reason. Perhaps a race condition, will have to test further.